CMakeLists.txt

```diff
@@ -16,32 +13,58 @@ if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Release)
 endif()
 
-option(SMF_INSTALL "Enable installation of smf" ON)
-option(SMF_ENABLE_TESTS "Useful for disabling all tests" ON)
+option(SMF_BUILD_PROGRAMS "Build smf programs and demos" ON)
 option(SMF_ENABLE_INTEGRATION_TESTS "control if integrations are bulit and ran" ON)
 option(SMF_ENABLE_UNIT_TESTS "control if unit tests are bulit and ran" ON)
 option(SMF_ENABLE_BENCHMARK_TESTS "control if benchmarks are bulit and ran" OFF)
-option(SMF_ENABLE_CMAKE_PROJECT_FLAGS "control cmake_cxx_flags_*" OFF)
-option(SMF_BUILD_PROGRAMS "Build smf programs and demos" ON)
+option(SMF_INSTALL "Enable installation of smf" ON)
+option(SMF_ENABLE_TESTS "Useful for disabling all tests" ON)
 
-if(NOT SMF_ENABLE_TESTS)
-  set(SMF_ENABLE_INTEGRATION_TESTS  OFF)
-  set(SMF_ENABLE_UNIT_TESTS  OFF)
-  set(SMF_ENABLE_BENCHMARK_TESTS  OFF)
+# fetch and build dependencies. each dependency is built in isolation allowing
+# tight control over the build configuration, and avoiding any inadvertant cmake
+# cache pollution. the dependencies are built at configure-time and then
+# imported as normal cmake dependencies, which will easily enable future
+# iterations to selectively satisfy dependencies from system packages.
+configure_file(CMakeLists.txt.in deps-build/CMakeLists.txt @ONLY)
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/deps-build)
+if(result)
+  message(FATAL_ERROR "CMake step for deps-build failed: ${result}")
+endif()
+execute_process(COMMAND ${CMAKE_COMMAND} --build .
+  RESULT_VARIABLE result
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/deps-build)
+if(result)
+  message(FATAL_ERROR "Build step for deps-build failed: ${result}")
 endif()
 
-include(ccache)
-include(doxygen_gen)
-include(packaging)
-include(set_option)
-include(tests)
-include(smfc_generator)
+# wire up dependency search paths
+foreach(dep_dir
+    "seastar_ext-prefix/src/seastar_ext/build/release"
+    "seastar_ext-prefix/src/seastar_ext/build/release/_cooking/installed"
+    "xxhash_ext-prefix"
+    "HdrHistogram_ext-prefix"
+    "gtest_ext-prefix"
+    "gflags_ext-prefix/lib/cmake"
+    "glog_ext-prefix/lib64/cmake"
+    "googlebenchmark_ext-prefix/lib/cmake")
+  list(APPEND CMAKE_PREFIX_PATH
+    "${CMAKE_CURRENT_BINARY_DIR}/deps-build/${dep_dir}")
+endforeach()
+
+# seastar includes several custom cmake modules
+list(APPEND CMAKE_MODULE_PATH
```

Comment:
nice! do we we want one file (.cmake) per dep, or one for all ? 

Having one per dep would match well w/ the `Find<dep>Config.cmake` and the rest of the convention. 

maybe we define a macro helper to write those `Manage<dep>.cmake` easier?